src/test/java/org/springframework/data/tarantool/repository/support/RestrictedUserIntegrationTest.java

```diff
@@ -0,0 +1,109 @@
+package org.springframework.data.tarantool.repository.support;
+
+import io.tarantool.driver.exceptions.TarantoolAccessDeniedException;
+import io.tarantool.driver.exceptions.TarantoolClientException;
+import io.tarantool.driver.mappers.MessagePackValueMapperException;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.dao.DataRetrievalFailureException;
+import org.springframework.data.mapping.MappingException;
+import org.springframework.data.tarantool.BaseIntegrationTest;
+import org.springframework.data.tarantool.RestrictedUserConfig;
+import org.springframework.data.tarantool.entities.SimpleObject;
+import org.springframework.data.tarantool.repository.SimpleObjectRepository;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * @author Artyom Dubinin
+ */
+@Tag("integration")
+@SpringBootTest(classes = RestrictedUserConfig.class)
```

Comment:
There's a simplier way to override confg values - see ```@TestPropertySource ```.
It' seems you don't need RestrictedUserConfig

Reply:
> Dynamic properties have higher precedence than those loaded from [@TestPropertySource](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/test/context/TestPropertySource.html)

https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/test/context/DynamicPropertySource.html


I also tried to use @DynamicPropertySource, but the values are overwritten in `BaseIntegrationTest` because the order is such that the child comes first and then the parent in the order of @DynamicPropertySource calls